tests/test_guard_diagnostics.py

```
@@ -0,0 +1,173 @@
+        diagnostic = CostGuard.to_diagnostic(result)
+        assert diagnostic.status is InfraDiagnosticStatus.BLOCKED
+        assert diagnostic.is_verified is False
+        assert diagnostic.proof_ref is None
+        assert diagnostic.developer_fields["audit_trace"]["rule_id"] == "COST_UNKNOWN_RESOURCE"
+
+
+class TestPydanticExtraForbid:
+    def test_iam_policy_rejects_extra(self):
+        from qwed_infra.guards.iam_guard import IamPolicy
+
+        with pytest.raises(Exception):
```

Ruff (0.15.18)

[warning] Do not assert blind exception: `Exception`

(B017)